components/datetime/tests/fixtures/tests/components-width-differences.json

```diff
@@ -15,6 +15,40 @@
             }
         }
     },
+    {
+        "description": "Width difference: Y vs YY",
+        "input": {
+            "value": "2020-01-07T08:25:07.000",
+            "options": {
+                "components": {
+                    "year": "TwoDigitWeekOf",
+                    "week": "NumericWeekOfYear"
+                }
+            }
+        },
+        "output": {
+            "values": {
+                "en": "week 2 of 20"
+            }
+        }
+    },
+    {
+        "description": "Width and symbol difference: y vs YY",
+        "input": {
+            "value": "2020-01-07T08:25:07.000",
+            "options": {
+                "components": {
+                    "year": "two-digit",
+                    "week": "NumericWeekOfYear"
+                }
+            }
+        },
+        "output": {
+            "values": {
+                "en": "week 2 of 20"
+            }
+        }
+    },
```

Comment:
Suggestion (optional): It seems possible to create strange combinations of week of year and other options in the components bag. It would be good to exercise these code paths with more robust tests. If we're not getting good results it might be worth filing bugs for them, or having `should_panic` tests.

Reply:
I've added a components-partial-matches fixture with a few combinations. This revealed that empty matches (e.g. by only asking for the week field) result in an empty string being rendered so I changed that case to return an error instead.

I also noticed that 'm' becomes 'm:ss' rather than failing to match. This is also the case upstream & the result is functional so I've just documented it with a test.